velox/vector/tests/VectorToStringTest.cpp

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "velox/vector/tests/VectorTestBase.h"
+
+namespace facebook::velox::test {
+
+class VectorToStringTest : public testing::Test, public VectorTestBase {};
+
+TEST_F(VectorToStringTest, flatIntegers) {
+  // No nulls.
+  auto flat = makeFlatVector<int32_t>({1, 2, 3, 4, 5, 6, 7, 8, 9, 10});
+  ASSERT_EQ(flat->toString(), "[FLAT INTEGER: 10 elements, no nulls]");
+  ASSERT_EQ(flat->toString(1), "2");
+  ASSERT_EQ(flat->toString(3, 8, ", ", false), "4, 5, 6, 7, 8");
+
+  // With nulls.
+  flat = makeFlatVector<int32_t>(
+      100, [](auto row) { return row; }, nullEvery(3));
+  ASSERT_EQ(flat->toString(), "[FLAT INTEGER: 100 elements, 34 nulls]");
+  ASSERT_EQ(flat->toString(1), "1");
+  ASSERT_EQ(flat->toString(33), "null");
+  ASSERT_EQ(flat->toString(0, 7, ", ", false), "null, 1, 2, null, 4, 5, null");
+}
+
+TEST_F(VectorToStringTest, arrayOfIntegers) {
+  // No nulls.
+  auto arr = makeArrayVector<int64_t>(
+      {{0}, {1, 2}, {3, 4, 5}, {6}, {}, {8, 9, 10, 11}});
+  ASSERT_EQ(arr->toString(), "[ARRAY ARRAY<BIGINT>: 6 elements, no nulls]");
+  ASSERT_EQ(arr->toString(3), "1 elements starting at 6 {6}");
```

Comment:
Its hard to understand what 6 represents here. It seems like an offset value.

If we have an array where there is a repetition of values, would this offset be any different ?

Reply:
sure